node/actors/bft/src/replica/replica_prepare.rs

```diff
@@ -0,0 +1,102 @@
+//! Handler of a ReplicaPrepare message.
+use super::StateMachine;
+use tracing::instrument;
+use zksync_concurrency::{ctx, error::Wrap};
+use zksync_consensus_roles::validator::{self};
+
+/// Errors that can occur when processing a "replica prepare" message.
+#[derive(Debug, thiserror::Error)]
+pub(crate) enum Error {
+    /// Message signer isn't part of the validator set.
+    #[error("Message signer isn't part of the validator set (signer: {signer:?})")]
+    NonValidatorSigner {
+        /// Signer of the message.
+        signer: validator::PublicKey,
+    },
+    /// Past view or phase.
+    #[error("past view/phase (current view: {current_view:?}, current phase: {current_phase:?})")]
+    Old {
+        /// Current view.
+        current_view: validator::ViewNumber,
+        /// Current phase.
+        current_phase: validator::Phase,
+    },
+    /// Invalid message signature.
+    #[error("invalid signature: {0:#}")]
+    InvalidSignature(#[source] anyhow::Error),
+    /// Invalid message.
+    #[error(transparent)]
+    InvalidMessage(validator::ReplicaPrepareVerifyError),
+    /// Internal error. Unlike other error types, this one isn't supposed to be easily recoverable.
+    #[error(transparent)]
+    Internal(#[from] ctx::Error),
+}
+
+impl Wrap for Error {
+    fn with_wrap<C: std::fmt::Display + Send + Sync + 'static, F: FnOnce() -> C>(
+        self,
+        f: F,
+    ) -> Self {
+        match self {
+            Error::Internal(err) => Error::Internal(err.with_wrap(f)),
+            err => err,
+        }
+    }
+}
+
+impl StateMachine {
+    #[instrument(level = "trace", skip(self), ret)]
+    pub(crate) async fn process_replica_prepare(
+        &mut self,
+        ctx: &ctx::Ctx,
+        signed_message: validator::Signed<validator::ReplicaPrepare>,
+    ) -> Result<(), Error> {
+        // ----------- Checking origin of the message --------------
```

Comment:
I think we can simplify handling this message to just extracting the high_qc and verifying it, while skipping everything else.

Reply:
Done.

Reply:
nit: I meant literally all validation, except for CommitQC.verify, and only after checking that this CommitQC has actually higher view than our high_qc, so that the signature validation is skipped entirely otherwise.